test/unit/puppetlabs/services/jruby_pool_manager/jruby_ref_pool_test.clj

```diff
@@ -23,6 +23,18 @@
                                           :max-borrows-per-instance max-borrows}
                                          options))))
 
+(deftest worker-id-persists
+  (testing "worker id is the same at borrow time and return time"
+    (jruby-testutils/with-pool-context
+      pool-context
+      jruby-testutils/default-services
+      (jruby-test-config 2)
+      (let [[instance borrowed-id] (pool-protocol/borrow pool-context)
+            returned-id (pool-protocol/return pool-context instance)]
+        (is (= (.getId (Thread/currentThread)) borrowed-id))
+        (is (= (.getId (Thread/currentThread)) returned-id))
+        (is (= borrowed-id returned-id))))))
```

Comment:
Probably don't need this last check now?

Reply:
eh yeah, I suppose